lib/mariaex/protocol.ex

```diff
@@ -295,63 +295,29 @@ defmodule Mariaex.Protocol do
   """
   def disconnect(_, state = %{sock: {sock_mod, sock}}) do
     msg_send(text_cmd(command: com_quit(), statement: ""), state, 0)
-    case msg_recv(state) do
-      {:ok, packet(msg: ok_resp()), _state} ->
-        sock_mod.close(sock)
-      {:ok, packet(msg: _), _state} ->
-        sock_mod.close(sock)
-      {:error, _} ->
-        sock_mod.close(sock)
-    end
-    _ = sock_mod.recv_active(sock, 0, "")
+    sock_mod.close(sock)
```

Comment:
This started to cause some troubles because there was no good way of knowing if `disconnect` is being called before or after the socket itself was actually closed. This created a difficult-to-debug situation where "left-over" data was stuck in the socket that the message decoder couldn't decode.

Then again, because the original code always closed the socket regardless of the actual response, it seems unnecessary to do the read, anyways.